airflow-core/src/airflow/utils/memray_utils.py

```diff
@@ -0,0 +1,69 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+import logging
+from collections.abc import Callable
+from functools import wraps
+from typing import ParamSpec, TypeVar
+
+from airflow.configuration import AIRFLOW_HOME, conf
+
+# Type variables for preserving function signatures
+PS = ParamSpec("PS")
+RT = TypeVar("RT")
+
+log = logging.getLogger(__name__)
+
+
+def enable_memray_trace(airflow_component_name: str) -> Callable[[Callable[PS, RT]], Callable[PS, RT]]:
```

Comment:
Ideally we should have an enum for the component names. I am not sure if our config supports list of enums natively for validation  - but even if not, it would be useful and we could add a feature to validate such enum list in config.